single_kernel_mongo/managers/cluster.py

```diff
@@ -229,6 +243,10 @@ def update_mongos_and_restart(self) -> None:
         key_file_contents = self.state.cluster.keyfile
         config_server_db_uri = self.state.cluster.config_server_uri
 
+        if ldap_user_to_dn_mapping := self.state.cluster.ldap_user_to_dn_mapping:
```

Comment:
will mongos be able to start without this data? 

another question - is this the best place in this file to save `ldap_user_to_dn_mapping ` to the peer databag? Where are `self.state.cluster.keyfile` and `self.state.cluster.config_server_uri` saved?

If this is the correct place to check - you will need a check for leadership since you are writing to the app data bag

Reply:
> will mongos be able to start without this data?

Yes, definitely it can :)

> another question - is this the best place in this file to save `ldap_user_to_dn_mapping ` to the peer databag? Where are `self.state.cluster.keyfile` and `self.state.cluster.config_server_uri` saved?

I'm storing it in the same location where it's stored on MongoDB.
I'm weighing however moving it to the ldap-peers relation databag to keep a better single responsibility principle of the databags.

> If this is the correct place to check - you will need a check for leadership since you are writing to the app data bag

That's very true, thanks a lot for the clear eyes ;)